features/directSecretProjection.feature

```
@@ -0,0 +1,171 @@
+                    apiVersion: v1
+                    kind: Secret
+                    name: $scenario_id-secret
+                workload:
+                    name: $scenario_id
+                    apiVersion: apps/v1
+                    kind: Deployment
+            """
+        Then Service Binding becomes ready
+        And jq ".status.binding.name" of Service Binding should be changed to "$scenario_id-secret"
+        And Content of file "/bindings/$scenario_id-binding/username" in workload pod is
+            """
+            foo
+            """
+        And Content of file "/bindings/$scenario_id-binding/password" in workload pod is
+            """
+            bar
+            """
+        And Content of file "/bindings/$scenario_id-binding/type" in workload pod is
+            """
+            db
+            """
+
+    Scenario: Bind a workload before the secret is created
+        Given Generic test application is running
+        And Service Binding is applied
+            """
+            apiVersion: servicebinding.io/v1beta1
+            kind: ServiceBinding
+            metadata:
+                name: $scenario_id-binding
+            spec:
+                service:
+                    apiVersion: v1
+                    kind: Secret
+                    name: $scenario_id-secret
+                workload:
+                    name: $scenario_id
+                    apiVersion: apps/v1
+                    kind: Deployment
+            """
+        When The Secret is present
+            """
+            apiVersion: v1
+            kind: Secret
+            metadata:
+                name: $scenario_id-secret
+            stringData:
+                username: foo
+                password: bar
+                type: db
+            """
+        Then Service Binding becomes ready
+        And jq ".status.binding.name" of Service Binding should be changed to "$scenario_id-secret"
+
+
+    Scenario: Fail to bind if the secret does not exist
```

I'm not sure this is a requirement of the spec. Can you cite the language that this test is based on?

---

The spec doesn't say anything about this case directly.  My interpretation comes from trying to maintain consistency with [provisioned services](https://github.com/servicebinding/spec/#ready-condition-status): if the service doesn't exist, we mark the binding as failed.  Similarly, if the secret doesn't exist, we won't resolve the binding.

Ultimately, this comes down to: what does an implementation do when the secret specified by a provisioned service doesn't exist?  As far as I can tell, there's no language in the spec on this.  Personally, I feel that saying the binding was successfully resolved hides an error from the user, since the workload didn't actually get any binding data.

Another interpretation would be to set the `ServiceAvailable` condition to `Unknown`, since we can technically mount the secret into the workload (even though the workload doesn't have binding data).  That said, I'm less certain on this interpretation.

---

An implementation can verify the existence of the Secret resource. But that's not a mandate or even a recommendation by the spec. So, I suggest not including this test in the conformance test suite.

However, if an implementation checks the Secret resource's existence, that implementation can add it as an acceptance test in their test suite.

Now there are pros and cons to both cases. If an implementation doesn't check the existence of the Secret resource, that implementation does not need to include RBAC rules for reading the Secret resource. From the Security perspective, that reduces the attack surface. But as you pointed out, if an implementation checks the existence of the Secret resource, it gives a good user experience by providing a status message.

---

As nice as would be, this probably should be something that we don't enforce via the specification.  _Requiring_ implementations to have permissions to read secrets doesn't seem like a good idea (though it may be something implementations wish to do for the reasons listed).

Removed in af3ab1d.